### PR TITLE
Update GitHub Actions testing workflow

### PR DESCRIPTION
**Pull request by AA-Turner**
Updated remaining jQuery scripts to standard JavaScript, and a few fixes for CSS changes.

I marked various backwards compatability points explicitly -- I don't know what the supported Sphinx versions are for this theme.

A

### CODE REVIEW

**Comment by JulienPalard** [comment]
This PR needs a review, it looks interesting. (Sadly my front-end skill don't allow me to understand it.)

**Comment by AA-Turner** [comment]
@pradyunsg would you have time for a review? (I admit I entirely forgot I made this PR...)

A

File: .github/workflows/tests.yml
Comment (hugovk) [inline .github/workflows/tests.yml:28]: Where is requests used?

```console
$ git grep requests
.github/workflows/tests.yml:          python -m pip install jinja2 markupsafe zc.lockfile requests
```

File: python_docs_theme/static/sidebar.js
Comment (hugovk) [inline python_docs_theme/static/sidebar.js:12]:
```suggestion
 * in .sphinxsidebar, after .sphinxsidebarwrapper, the #sidebarbutton
```

File: python_docs_theme/theme.conf
Comment (hugovk) [inline python_docs_theme/theme.conf:17]: Consistency:
```suggestion
sidebarbtncolor = #cccccc
```

File: .github/workflows/tests.yml
Comment (AA-Turner) [inline .github/workflows/tests.yml:28]:
```suggestion
          python -m pip install jinja2 markupsafe zc.lockfile
```

Comment (AA-Turner) [reply .github/workflows/tests.yml]: image

This must be why...

File: .github/workflows/tests.yml
Comment (hugovk) [inline .github/workflows/tests.yml:22]: Ah right! Let's use the existing requirements.txt from that repo to avoid duplication and needing to keep both in sync:

```suggestion
          python -m pip install -r docsbuild-scripts/requirements.txt
```

**Comment by AA-Turner** [comment]
@JulienPalard can we build and publish a test version of CPython's docs with this PR? That might assuage your concerns over the JS.

A

File: .github/workflows/tests.yml
Comment (JulienPalard) [reply .github/workflows/tests.yml]: > Where is requests used?

To issue PURGE "commands" to the CDN.

**Comment by JulienPalard** [comment]
> @JulienPalard can we build and publish a test version of CPython's docs with this PR? That might assuage your concerns over the JS.

Yes but it will be overwritten by the daily build so I doubt it's a good idea. However a build is available in the "checks" tab as a build artifact, so we can test locally. It is done using the same script as in production so it should be similar.

File: .github/workflows/tests.yml
Comment (hugovk) [inline .github/workflows/tests.yml:37]: Let's bump this, and use just major like the others:
```suggestion
        uses: actions/upload-artifact@v3
```

**Review by pradyunsg — APPROVED** [review]

File: .github/workflows/tests.yml
Comment (ezio-melotti) [inline .github/workflows/tests.yml:20]:
```suggestion
      - name: Setup requirements
```

File: .github/workflows/tests.yml
Comment (ezio-melotti) [inline .github/workflows/tests.yml:35]: Should this be `3.11` now?
```suggestion
          --branch 3.11
```

**Review by ezio-melotti — APPROVED** [review]

File: .github/workflows/tests.yml
Comment (hugovk) [inline .github/workflows/tests.yml:37]: This was accidentally reverted in a force push:
```suggestion
        uses: actions/upload-artifact@v3
```

File: .github/workflows/tests.yml
Comment (hugovk) [inline .github/workflows/tests.yml:20]: Nit: "setup" is a noun, "set up" for the verb form:
```suggestion
      - name: Set up requirements
```

File: .github/workflows/tests.yml
Comment (hugovk) [inline .github/workflows/tests.yml:35]: Much more readable! 👍

**Review by hugovk — APPROVED** [review]
Thanks!

**Comment by hugovk** [comment]
Let's merge, this only updates the GitHub Actions workflow file and importantly allows testing feature branches (I want to test some changes before creating a PR), and has three approvals.

Thanks!

**Pull request merged**
